mainwindow.cc

```diff
@@ -3723,6 +3762,21 @@ void MainWindow::unzoom()
   applyZoomFactor();
 }
 
+void MainWindow::viewLinkClicked( const QUrl & url )
+{
+  if( scanPopup.get() && scanPopup->isActiveWindow() )
```

Comment:
> I follow the rule if it not works well,fix it,goldendict(or you) follow the rule if it works ,do not fix it .
that cause all the issues here.

The `isActiveWindow` workaround is ridiculous. It might be acceptable in a temporary hotfix for a stable version, but not in an initial Qt WebEngine design. It is one consequence of haphazard design, which likely introduces many more regressions. If an alternative approach allows to improve the design and get rid of workarounds like this, then it should be at least considered and compared to the current implementation.